wptherml/em.py

```diff
@@ -1013,7 +1021,228 @@ def compute_pv_stpv(self):
+            )  # Integrate for short circuit current
+
+        # go back to original spectrum
+        self.remove_layer(_ln)
+        self.compute_spectrum()
+        self.compute_spectrum_gradient()
+    # Other figure of merit calculations here to be called in compute_pv_stpv
+
+    def compute_pv_stpv_total_incident_power(self):
+        """Docstring
+        Use equation npv = Jsc * Voc * FF
+        Jsc = short circuit current
+        Voc = open circuit current
+        FF = fill factor/ratio of ontainable power to short circuit * open circuit voltage
+        
+        The plan:
+
+        initialze npv (assuming a static number), calculate Voc and FF, and multiply these three together to get total incident power as a unitless efficiency 
+        Voc = (kB*Temperature/charge)*(ln(short circuit current)/(initial current))
+        total_incident_power = pv_stpv_short_circuit_current_gradient * Voc * Fill factor
+        
+        """
+        pass 
+    
+    def compute_pv_stpv_splitting_power(self):
+        
+        """  
+        Docstring
+
+        Method to compute the pv_stpv splitting power as defined in Eq. (46) of https://www.overleaf.com/project/648a0cfeae29e31e10afc075 
+        Attributes
+        -----------
+        emissivity_AR_polystyrene : array
+                                    Storage of emissivity array.
+        sliced_wavelength_array : array
+                                    Slice of the wavelength array over the upper and lower limits determined by the minimum and maximum
+                                    differences of array values and the integral's bounds.
+        sliced_emissivity_array : array
+                                    Slice of the emissivity array over the upper and lower limits determined by the minimum and maximum
+                                    differences of array values and the integral's bounds.
+        pv_stpv_splitting_power : array
+                                    Emissivity array integrated over 3.0 to 3.5 microns.
+        
+        Returns
+        -------
+        None
+
+        Notes:  The emissivity needs to be computed for the reversed original stack (meaning the stack *without the active layer*) before updating the 
+                thermal emission spectrum.
+                Steps:
+                1. Reverse the stack
+                2. Compute the optical spectra
+                3. Compute the thermal emission spectra
+                4. Define the integrand in Eq. (46)
+                5. Integrate the integrand and store to the attribute self.pv_stpv_splitting_power
+        
+        """
+               
+        # Reverse stack, active layer was removed in the last function
+        # Compute the optical and thermal spectra
+        self.reverse_stack()
+        self.compute_spectrum()
+        self._compute_therml_spectrum()
+        
+    
+        # Store spectra 
+        emissivity_AR_polystyrene = self.emissivity_array
```

Comment:
It is safer to copy numpy arrays using np.copy() rather than assign them using just an equals sign.

You can do this with the line
emissivity_AR_polystyrene = np.copy(self.emissivity_array)

Reply:
You may (optionally) delete the lines where the transmissivity and reflectivity arrays are copied because we do not use those in this method